Tests/NetworkServiceTests/NetworkErrorTests/NetworkErrorTests.swift

```diff
@@ -0,0 +1,29 @@
+import NetworkService
+import XCTest
+
+final class NetworkErrorTests: XCTestCase {
+    func testAllNetworkErrorsExistInAPI() {
+        // Given
+        let possibleNetworkErrors = Set(NetworkError.allCases)
+
+        // When
+        let expectedNetworkErrors: Set = [
+            NetworkError.lostConnection,
```

Comment:
TBH, It's not clear for me why these cases are chosen and covered. Seems we can start with the following cases within the `NetworkError`:
```swift
    /// ...
    case badRequest

    /// Indicates that the client request has not been completed
    /// because it lacks valid authentication credentials for the requested resource.
    case authenticationError

    /// Indicates the server cannot or will not process the request
    /// due to something that is perceived to be a client error.
    case clientError

    /// Indicates the server cannot or will not process the request due to server issue.
    case serverError

    /// ...
    case unknownError

    /// ...
    case requestTimeout
```

WDYT?

Reply:
I think it's a good idea because I didn't quite understand some of the cases myself.